bazel/repository_locations.bzl

```diff
@@ -282,8 +282,9 @@ REPOSITORY_LOCATIONS = dict(
     ),
     com_googlesource_googleurl = dict(
         # Static snapshot of https://quiche.googlesource.com/quiche/+archive/googleurl_dbf5ad147f60afc125e99db7549402af49a5eae8.tar.gz
-        sha256 = "b40cd22cadba577b7281a76db66f6a66dd744edbad8cc2c861c2c976ef721e4d",
-        urls = ["https://storage.googleapis.com/quiche-envoy-integration/googleurl_dbf5ad147f60afc125e99db7549402af49a5eae8.tar.gz"],
+        sha256 = "5bdbecce22d522f03ac86c6a694d5ff76780a16d6d682910e87ac156b0a25b21",
+        strip_prefix = "googleurl-dbf5ad1",
+        urls = ["https://github.com/dio/googleurl/archive/dbf5ad1.tar.gz"],
```

Comment:
Temporary fix: since it has a top-level `BUILD` file and a `build` directory (which is problematic when extracting on mac and windows).

Reply:
I assume can just patch?

Reply:
Seems like we can't, since it is failing when `http_archive` extracting the tarball, it is stopped when trying to extract `BUILD` when it saw `build` dir already there. I'm not sure how we can have a patch on the downloaded tarball before extracting.

Reply:
Probably @danzh2010 could shed some light here.

Reply:
> A temporary fix, since it has a top-level `BUILD` file and a `build` directory (which is problematic when extracting on mac and windows). I renamed the `BUILD` file to `BUILD.bazel`, I think removing it is OK.

The top level BUILD file is a dummy but needed for the build_config.bzl to setup default copts. But they probably can be moved to a subdirectory instead.

Reply:
As to `build` directory, it is from chromium, so nothing I can do there. What's the exact issue of having it under root directory?

Reply:
@danzh2010 with the original archive, `http_archive` tries to extract `BUILD` file while the `build` directory is already there. For case-insensitive filesystem like Windows and macOS it is problematic.

Reply:
If I move the top level BUILD file and build_config.bzl to some sub-directory, would it help your case?

Reply:
Yes! Thanks @danzh2010!

Reply:
@danzh2010 sorry. Can we still pick this up? (moving the top-level BUILD file)